packages/common/src/AncillaryDataParser.ts

```diff
@@ -0,0 +1,254 @@
+import Web3 from "web3";
+
+type CharObject = {
+  character: string;
+  escape: boolean;
+  skip: boolean;
+};
+
+/**
+ * @title Parse ancillary data.
+ * @notice Ancillary data parser implementation following guidelines at:
+ * https://docs.google.com/document/d/1zhKKjgY1BupBGPPrY_WOJvui0B6DMcd-xDR8-9-SPDw/edit
+ * @param {String} hex string representation of ancillaryData
+ * @return {Object} parsed ancillary data object.
+ */
+export function parseAncillaryData(ancillaryData: string): Record<string, unknown> {
+  // Some requesting contracts set the synthetic token address as ancillary data, so try to parse it first:
+  if (Web3.utils.isAddress(ancillaryData)) return { address: ancillaryData };
+  let ancillaryString;
+  try {
```

Comment:
should this throw if it cant use `Web3.utils.hexToUtf8`? it seems like if it is an `AncillaryString` it's actually a pretty normal flow and should not throw getting here.